benchmark/benchmark.cpp

```diff
@@ -0,0 +1,75 @@
+#include <chrono>
+#include <iostream>
+#include <iomanip>
+#include <locale>
+
+#include <benchmark.hpp>
+
+namespace benchmark {
+
+// update this map when a new feature is supported
+// key is the name and value is the function doing benchmarking
+std::map<std::string, std::function<void()>> features {
+   { "alt_bn_128", alt_bn_128_benchmarking },
+   { "modexp", modexp_benchmarking },
+   { "key", key_benchmarking },
+   { "hash", hash_benchmarking },
+};
+
+// values to control cout format
+constexpr auto name_width = 28;
+constexpr auto runs_width = 5;
+constexpr auto time_width = 13;
+constexpr auto ns_width = 2;
+
+uint32_t num_runs = 1;
+
+std::map<std::string, std::function<void()>> get_features() {
+   return features;
+}
+
+void set_num_runs(uint32_t runs) {
+   num_runs = runs;
+}
+
+void printt_header() {
+   std::cout << std::left << std::setw(name_width) << "function"
```

Comment:
Is this supposed to be `printt_header` or `print_header`?

Reply:
Thanks for your sharp eyes! Sorry for this copy/paste error.